File: clinicapro-spa/src/app/core/models/paciente.ts
import {Pessoa} from "./pessoa";
import {FamiliarPaciente} from "./FamiliarPaciente";
import {TipoCaptacaoEnum} from "./enums/tipoCaptacaoEnum";

export class Paciente extends Pessoa {
    familiares: FamiliarPaciente[] = [];
    tipoCaptacao: TipoCaptacaoEnum;

    constructor(paciente?: Partial<Paciente>) {
        super(paciente);
        if (paciente) {
            Object.assign(this, paciente);
        }
    }
}

File: clinicapro-api/src/main/java/br/com/clinicapro/api/domain/Paciente.java
package br.com.clinicapro.api.domain;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.OneToMany;

import java.util.List;

@Entity
public class Paciente extends Pessoa {

    private String tipoCaptacao;

    @OneToMany(mappedBy = "paciente", cascade = CascadeType.ALL)
    private List<FamiliarPaciente> familiares;

    public String getTipoCaptacao() {
        return tipoCaptacao;
    }

    public void setTipoCaptacao(String tipoCaptacao) {
        this.tipoCaptacao = tipoCaptacao;
    }

    public List<FamiliarPaciente> getFamiliares() {
        return familiares;
    }

    public void setFamiliares(List<FamiliarPaciente> familiares) {
        this.familiares = familiares;
    }
}
